src/config/mod.rs

```diff
@@ -1552,14 +1552,8 @@ impl DbConfig {
             )
             .into());
         }
-        if self.max_sub_compactions == 0
-            || self.max_sub_compactions as i32 > self.max_background_jobs
-        {
-            return Err(format!(
-                "max_sub_compactions should be greater than 0 and less than or equal to {:?}",
-                self.max_background_jobs,
-            )
-            .into());
+        if self.max_sub_compactions == 0 {
+            return Err("max_sub_compactions should be greater than 0".into());
```

Comment:
I think it's better to log a message and reset self.max_sub_compactions = self.max_background_jobs when self.max_background_jobs < self.max_sub_compactions.

Reply:
Those two are actually unrelated. sub compactions are scheduled in temporary threads instead of the shared threadpool.